Allow tying jumpsuits around waists (Impstation port)

alterae opened the pull request:
Port of https://github.com/impstation/imp-station-14/pull/375

## About the PR
Allows (some) jumpsuits to be tied around the waist. Mostly upstream ones. There are some weird omissions.

Notable jumpsuits NOT made foldable include:
- The Salvage Specialist and HoP jumpsuits, which use _DV sprites at the moment.
- Numerous security jumpsuits (due to rollable sleeves).
- Pretty much all the colored jumpsuits for some reason.
- The prisoner jumpsuit (this one in particular feels like a huge oversight).
- All DeltaV-, Nyano-, EE-, and Impstation-specific jumpsuits.
- Several seemingly random upstream jumpsuits including the reporter's jumpsuit and all variant bartender uniforms.
- All skirts.

> [!IMPORTANT]
> I will not be doing any spritework for this PR myself, but will GLADLY accept additional sprites and do the YAML for them.

## Why / Balance
Butch lesbians of the world rejoice (especially now that we have undershirts).

## Technical details
Adds two new locale keys and one new YAML prototype. Modifies nearly forty upstream YAML prototypes and nearly forty RSI `meta.json`s. Adds nearly forty new sprite PNGs. Does not conflict with the existing mechanism for rolling up the sleeves on certain security uniforms.

## Media
<img width="177" alt="image" src="https://github.com/user-attachments/assets/35cb07e2-d6e5-43bb-89dc-571b54e6f615" />
<img width="207" alt="image" src="https://github.com/user-attachments/assets/c6f455fd-61d0-4b96-a3ea-25c04a9ead69" />
<img width="239" alt="image" src="https://github.com/user-attachments/assets/efdb3d37-b10f-4ad5-b73a-11c3c60dc9f7" />
<img width="151" alt="image" src="https://github.com/user-attachments/assets/e3627cee-82f0-44e4-9b4b-76b073c8fa4f" />
<img width="170" alt="image" src="https://github.com/user-attachments/assets/b0dead14-d82e-4d56-b38f-ca696d37c9e6" />


## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
Some prototypes have an additional parent. I don't think there's any actual work needed to deal with this.

**Changelog**
:cl:
- add: Ported the ability to tie (some) jumpsuit jackets around your waist from Impstation. Jumpskirts not included

alterae committed:
Make (some) jumpsuits foldable

Still porting impstation/imp-station-14#375. Notable jumpsuits NOT made
foldable include:
- The Salvage Specialist and HoP jumpsuits, which use _DV sprites at
  the moment.
- Numerous security jumpsuits (due to rollable sleeves).
- Pretty much all the colored jumpsuits for some reason.
- The prisoner jumpsuit (this one in particular feels like a huge
  oversight).
- All DeltaV-, Nyano-, EE-, and Impstation-specific jumpsuits.
- Several seemingly random upstream jumpsuits including the reporter's
  jumpsuit and all variant bartender uniforms.
- All skirts.

Uncertain at this time whether to let those come in a separate PR.

Radezolid commented on Resources/Locale/en-US/_Impstation/foldable/components/foldable-component.ftl:
Add an EoF newline.

Radezolid commented on Resources/Prototypes/_Impstation/Entities/Clothing/Uniforms/base_clothinguniforms.yml:
Add an EoF newline.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/atmos.rsi/meta.json:
```diff
@@ -1,8 +1,7 @@
   "license": "CC-BY-SA-3.0",
-  "copyright": "https://github.com/tgstation/tgstation/commit/beaea876ea426c0e215cee64619862dc19bd9cd8, monkey made by brainfood1183 (github)",
-  "size": {
+  "copyright": "https://github.com/tgstation/tgstation/commit/beaea876ea426c0e215cee64619862dc19bd9cd8, monkey made by brainfood1183 (github), folded version taken from tgstation and edited by Pinkbat5 (github)",  "size": {
```
Fix this, size should be under copyright.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/atmos.rsi/meta.json:
```diff
@@ -27,4 +30,4 @@
   ]
-}
\ No newline at end of file
+}
```
Sadly untouch this please.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/atmos_casual.rsi/meta.json:
```diff
@@ -1,8 +1,7 @@
   "license": "CC-BY-SA-3.0",
-  "copyright": "Sprited by KingFroozy (github) for space station 14",
-  "size": {
+  "copyright": "Sprited by KingFroozy (github) for space station 14, folded version taken from tgstation and edited by Pinkbat5 (github)",  "size": {
```
Fix the size being in the wrong place.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/brigmedic.rsi/meta.json:
```diff
@@ -1,8 +1,7 @@
   "license": "CC-BY-SA-3.0",
-  "copyright": "Made by PuroSlavKing (github) for SS14 | Resprited by Hülle#2562 (Discord), monkey made by brainfood1183 (github) for ss14",
-  "size": {
+  "copyright": "Made by PuroSlavKing (github) for SS14 | Resprited by Hülle#2562 (Discord), monkey made by brainfood1183 (github) for ss14, folded version taken from tgstation spritework and edited by Pinkbat5 (github)",  "size": {
```
Same as before

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/cargotech.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039, edited by Flareguy. In hand sprite scaled down by potato1234_x, monkey made by brainfood1183 (github) for ss14",
+  "copyright": "Taken from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039, edited by Flareguy. In hand sprite scaled down by potato1234_x. Folded version taken from tgstation and edited by Pinkbat5 (github)",
```
Fix the removal of attribution.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/hos.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039. In hand sprite scaled down by potato1234_x, monkey made by brainfood1183 (github) for ss14",
+  "copyright": "Created by Flareguy using sprites from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039 & the jumpsuit base found in color.rsi. Folded version taken from tgstation spritework and edited by Pinkbat5 (github)",
```
Fix the removal of attribution.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/roboticist.rsi/meta.json:
```diff
@@ -27,4 +31,4 @@
   ]
-}
\ No newline at end of file
+}
```
untouch

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/security.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039. In hand sprite scaled down by potato1234_x, monkey made by brainfood1183 (github) for ss14. equipped-INNERCLOTHING edited by Dutch-VanDerLinde (github).",
+  "copyright": "Created by Flareguy using sprites from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039 & the jumpsuit base found in color.rsi. Folded version taken from tgstation and edited by Pinkbat5 (github)",
```
Fix removal of attribution and the adding of Flareguy sprites attribution.

Radezolid commented on Resources/Textures/Clothing/Uniforms/Jumpsuit/warden.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039. In hand sprite scaled down by potato1234_x, monkey made by brainfood1183 (github) for ss14",
+  "copyright": "Created by Flareguy using sprites from tgstation at commit https://github.com/tgstation/tgstation/commit/c838ba21dae97db345e0113f99596decd1d66039 & the jumpsuit base found in color.rsi. Folded version taken from tgstation spritework and edited by Pinkbat5 (github)",
```
Same as before

alterae committed:
Add foldable uniform jumpsuit base component

Initial work on porting impstation/imp-station-14#375.
Component gets the Impstation suffix to not conflict with the existing
DeltaV component for rolled-up sleeves on some security uniforms.

Lyndomen reviewed (dismissed):
Lots of spriting work on the horizon!

alterae commented:
> Lots of spriting work on the horizon! 

does this constitute Direction approval or just lyndo approval?

SolStar2 commented:
> does this constitute Direction approval or just lyndo approval?

Direction approved 👍

alterae commented:
> Direction approved 👍 

epic
gimme about 4.5–5hrs to fix up the attribution like Razedolid wanted and then this should be ready for merge

alterae commented:
Radezolid's desired attribution fix has been completed (as discussed on the Discord) so this _should_ be ready for merge now…

deltanedas reviewed (changes requested):
for the meta.json put deltav specific stuff after a `|` not `,` or `.`


like `monkey made by brainfood1183 (github) for ss14 | Folded version taken`

alterae committed:
Add folded jumpsuit sprites for (some) upstream jumpsuits

Once again ported from impstation/imp-station-14#375. Only some of our
many, many jumpsuits are affected, and all of them are in the default
namespace.

Some of these sprites aren't even used because our own (unmodified) _DV
sprites take priority, so there is a lot of cleanup to be done.
